2 dz/2/main.cpp

```diff
@@ -0,0 +1,35 @@
+/*Реализовать возведение в целую степень (с логарифмической сложностью алгоритма).
+
+*/
+
+#include <iostream>
+
+using namespace std;
+
+double power(double a, int n)
+{
+    double result = 1;
+    while(n > 0)
+    {
+        if (n & 1)
+        {
+            result *= a;
+        }
+        n = n >> 1;
+        a *= a;
+    }
+    return result;
+}
+
+int main()
+{
+    double the_basis_of_the_degree = 0;
```

Comment:
It's a snake_case, but in this project, we decided to use CamelCase for classes/structures and camelCase for variable and function names :)
Style guide p. 11

Reply:
Probably the names "base" and "exponent". Or "degreeBasis" and "degreeIndicator".
We usually don't use articles, conjunction, prepositions in variable names.